.travis.yml

```diff
@@ -22,3 +22,7 @@ install:
 script:
   - npm run doc
   - npm run test
+
+cache:
+  directories:
+    - "node_modules"
```

Comment:
This is not a good idea, because strapi is using some binary dependencies like node-gyp that needs to be rebuilt if the node version has changed.

Reply:
This is no issue as this would invalidate the cache. If this would be really an isue we can change it to the npm cache.

Reply:
Unfortunately, It doesn't.

Reply:
See https://stackoverflow.com/questions/42521884/should-i-have-travis-cache-node-modules-or-home-npm, the second approach never caused any issues so far and we can use it if the first approach produces any issues.

Reply:
But so far there are other issues which we have to resolve first: https://travis-ci.org/strapi/strapi/builds/389204771#L500

Reply:
I think that's safer to change it to npm cache rather than node_modules, I'm afraid that caching node_mogules will cause some problems in the future and it going to be really hard to debug, thanks for the investigation BTW :+1: